zlib-rs/src/inflate.rs

```diff
@@ -517,6 +517,348 @@ const INFLATE_FAST_MIN_HAVE: usize = 15;
 const INFLATE_FAST_MIN_LEFT: usize = 260;
 
 impl State<'_> {
+    // This logic is split into its own function for two reasons
+    //
+    // - We get to load state to the stack; doing this in all cases is expensive, but doing it just
+    //      for Len and related states is very helpful.
+    // - The `-Cllvm-args=-enable-dfa-jump-thread` llvm arg is able to optimize this function, but
+    //      not the entirity of `dispatch`. We get a massive boost from that pass.
+    //
+    // It unfortunately does duplicate the code for some of the states; deduplicating it by having
+    // more of the states call this function is slower.
```

Comment:
I don't think it is because of the labels that we break/continue too. Also it's _fine_ because we won't touch this much ever again hopefully. But not ideal for sure.

Reply:
You can pass the labels as macro arguments, right?

Reply:
can you? what would be the fragment specifier of the label? is it an `identifier` somehow? `tt` might work but often requires brackets

Reply:
also in our function we load the reader and writer to the stack. you could parameterize the macro on that too but, idk, is that worth it?

Reply:
At the very least please add a comment to every copy of every state that is duplicated indicating that they should be kept in sync.